syntax = "proto3";

package graph_serialize;

message Edge {
    bytes edgeName = 1;
    int32 spanCount = 2;
    int32 from = 3;
    int32 to = 4;
    double weight = 5;
}

message Vertex {
    repeated int32 edgeId = 1;
}

message Graph {
    repeated Edge edge = 1;
    repeated Vertex vertex = 2;
}